torch/_inductor/codegen/triton.py

```diff
@@ -217,13 +226,20 @@ def boundary_check(self) -> List[int]:
         """List of indices to pass to tl.load(boundary_check=...)"""
         check = []
         for i in range(len(self.shape)):
+            # Look up the block prefix. For a string like "XBLOCK", this is "X".
+            # If the shape is a sympy expression, skip this check.
+            try:
+                block_prefix = self.block_shape[i][0]  # type: ignore[arg-type]
+            except TypeError:
+                block_prefix = None
+
             if (
                 self.block_shape[i] != "1"
```

Comment:
Should this become `statically_known_equals`, now that `block_shape` can be an expression?